controllers/discovery/initDiscoveryUserData.js

```diff
@@ -76,12 +136,15 @@ const InitDiscoveryUserData = async (req, res) => {
         ON CommonUsers.user_id = A.user_id
         LEFT JOIN user_follow_user B
         ON A.user_id = B.user_id_followed AND B.user_id_follower = :userId
-        WHERE A.user_id != :userId AND A.is_anonymous = false AND A.is_banned = false
+        WHERE 
+          A.user_id != :userId 
+          AND A.is_anonymous = false 
+          AND A.is_banned = false
+          ${where_anon_dm} 
         ORDER BY
-        COALESCE(B.user_id_follower, '') DESC,
-        COALESCE(A.karma_score, 0) DESC,
-        COALESCE(CommonUsers.common, -1) DESC, 
-        COALESCE(CommonUsers.user_match, -1) DESC
+          is_followed DESC,
+          followersCount DESC,
+          COALESCE(A.karma_score, 0) DESC
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `is_followed` and `followersCount` fields are not defined in the SQL query. They should be included in the SELECT clause to avoid a SQL error.
```diff
-         WHERE 
-           A.user_id != :userId 
-           AND A.is_anonymous = false 
-           AND A.is_banned = false
-           ${where_anon_dm} 
-          ORDER BY
-           is_followed DESC,
-           followersCount DESC,
-           COALESCE(A.karma_score, 0) DESC
+         WHERE 
+           A.user_id != :userId 
+           AND A.is_anonymous = false 
+           AND A.is_banned = false
+           ${where_anon_dm} 
+          ORDER BY
+           (SELECT COUNT(*) FROM user_follow_user WHERE user_id_followed = A.user_id) DESC,
+           EXISTS (
+               SELECT 1
+               FROM user_follow_user AS f2
+               WHERE f2.user_id_follower = :userId
+               AND f2.user_id_followed = A.user_id
+           ) DESC,
+           COALESCE(A.karma_score, 0) DESC
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->